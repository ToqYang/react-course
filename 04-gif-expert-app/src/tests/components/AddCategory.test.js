import React from "react";
import AddCategory from "../../components/AddCategory";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";

describe("Pruebas en Add Category", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("Mostrarse Correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe cambiar el texto", () => {
    const input = wrapper.find("input");
    const value = "Hello World";
    input.simulate("change", { target: { value } });
    expect(wrapper.find("p").text().trim()).toBe(value);
  });

  test("No debe postear", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Debe cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hello";
    input.simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(input.text()).toBe("");
  });
});
