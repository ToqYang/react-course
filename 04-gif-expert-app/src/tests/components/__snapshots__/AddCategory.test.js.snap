// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Add Category Mostrarse Correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
